app-crypt/johntheripper/johntheripper-1.8.0.recipe

```diff
@@ -0,0 +1,188 @@
+SUMMARY="A fast password cracker"
+DESCRIPTION="A fast password cracker, currently available for many flavors of \
+Unix, Windows, DOS, and OpenVMS. Its primary purpose is to detect weak Unix \
+passwords. Besides several crypt(3) password hash types most commonly found on \
+various Unix systems, supported out of the box are Windows LM hashes, plus \
+lots of other hashes and ciphers in the community-enhanced version."
+HOMEPAGE="http://www.openwall.com/john"
+COPYRIGHT="1996-2013 Solar Designer
+	2014 John the Ripper '-jumbo' authors"
+LICENSE="GNU GPL v2
+	John the Ripper"
+REVISION="1"
+jumboVersion="1"
+SOURCE_URI="http://www.openwall.com/john/j/john-$portVersion-jumbo-$jumboVersion.tar.xz"
+CHECKSUM_SHA256="bac93d025995a051f055adbd7ce2f1975676cac6c74a6c7a3ee4cfdd9c160923"
+SOURCE_DIR="john-$portVersion-jumbo-$jumboVersion"
+# Upstream patches, can be dropped in the next version
+PATCHES="johntheripper-$portVersion-upstreamed.patchset
+	johntheripper-$portVersion.patchset"
+
+ARCHITECTURES="!x86_gcc2 x86_64"
+SECONDARY_ARCHITECTURES="x86"
+
+PROVIDES="
+	johntheripper$secondaryArchSuffix = $portVersion
+	cmd:john = $portVersion
+	"
+
+bins=(calc_stat cprepair genmkvpwd luks2john mkvcalcproba raw2dyna
+	tgtsnarf uaf2john wpapcap2john)
+syms=(base64conv dmg2john gpg2john hccap2john keepass2john keychain2john
+	keyring2john keystore2john kwallet2john pfx2john putty2john pwsafe2john
+	racf2john rar2john ssh2john truecrypt_volume2john unafs undrop unique
+	unshadow zip2john)
+
+for b in ${bins[@]} ${syms[@]}
+do
+	PROVIDES+="
+		cmd:$b = $portVersion
+		"
+done
+
+REQUIRES="
+	haiku$secondaryArchSuffix
+	lib:libcrypto$secondaryArchSuffix
+	lib:libgmp$secondaryArchSuffix
+	lib:libgomp$secondaryArchSuffix
+	lib:libz$secondaryArchSuffix
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	devel:libcrypto$secondaryArchSuffix
+	devel:libgmp$secondaryArchSuffix
+	devel:libz$secondaryArchSuffix
+"
+BUILD_PREREQUIRES="
+	cmd:awk
+	cmd:find
+	cmd:gcc$secondaryArchSuffix
+	cmd:grep
+	cmd:make
+	cmd:sed
+	"
+
+GLOBAL_WRITABLE_FILES="
+	settings/john/john.conf keep-old
```

Comment:
(Oops, I wanted to leave a comment yesterday about the "keep-old" but then forgot to do so!)
There is a special keyword, "auto-merge", which might be worth using here.
```diff
-	settings/john/john.conf keep-old
+	settings/john/john.conf auto-merge
```
I've never witnessed this feature in action but I guess it would be appropriate to use it here, since this is a text file.